src/ffi/macros.rs

```diff
@@ -17,7 +17,43 @@ macro_rules! ffi_fn {
         }
     };
 
+    ($(#[$doc:meta])* fn $name:ident($($arg:ident: $arg_ty:ty),*) -> $ret:ty $body:block ?= $default:expr) => {
+        $(#[$doc])*
+        #[no_mangle]
+        pub extern fn $name($($arg: $arg_ty),*) -> $ret {
+            use std::panic::{self, AssertUnwindSafe};
+
+            match panic::catch_unwind(AssertUnwindSafe(move || $body)) {
+                Ok(v) => v,
+                Err(_) => {
+                    eprintln!("panic unwind caught, returning default");
```

Comment:
Yea, you may be right. I'm kinda thinking that if we've caught and properly returned a value, perhaps we shouldn't be trying to print to stdio...

Reply:
I'll remove it.